hs_discover/views.py

```diff
@@ -1,57 +1,224 @@
+            subjects = sqs.facet_counts()['fields']['subject'][:self.filterlimit]
+            sqs = SearchQuerySet().facet('contributor')
+            contributors = sqs.facet_counts()['fields']['contributor'][:self.filterlimit]
+            sqs = SearchQuerySet().facet('resource_type')
+            types = sqs.facet_counts()['fields']['resource_type'][:self.filterlimit]
+            sqs = SearchQuerySet().facet('availability')
+            availability = sqs.facet_counts()['fields']['availability'][:self.filterlimit]
+
+            return Response({
+                'time': (time.time() - start),
+                'filterdata': json.dumps([authors, owners, subjects, contributors, types, availability])
+            })
+
+        asc = '-1'
+        if request.GET.get('asc'):
+            asc = request.GET.get('asc')
+
+        sort = 'modified'
+        if request.GET.get('sort'):
+            sort = request.GET.get('sort')
+        sort = sort if asc == '1' else '-{}'.format(sort)
+
+        if request.GET.get('q'):
+            q = request.GET.get('q')
+            sqs = sqs.filter(content=q)  # .boost('keyword', 2.0)
 
-        sqs = SearchQuerySet()
+        try:
+            qs = request.query_params
+            filters = json.loads(qs.get('filter'))
+            # filter values expect lists, for example discoverapi/?filter={"owner":["Firstname Lastname"]}
+            if filters.get('author'):
+                sqs = sqs.filter(author__in=filters['author'])
+            if filters.get('owner'):
+                sqs = sqs.filter(owner__in=filters['owner'])
+            if filters.get('subject'):
+                sqs = sqs.filter(subject__in=filters['subject'])
+            if filters.get('contributor'):
+                sqs = sqs.filter(contributor__in=filters['contributor'])
+            if filters.get('type'):
+                sqs = sqs.filter(resource_type__in=list(filters['type']))
+            if filters.get('availability'):
+                sqs = sqs.filter(availability__in=filters['availability'])
+            if filters.get('geofilter'):
+                sqs = sqs.filter(north__range=[-90, 90])
+            if filters.get('date'):
+                try:
+                    datefilter = DateRange(start=datetime.datetime.strptime(filters['date'][0], '%Y-%m-%d'),
+                                           end=datetime.datetime.strptime(filters['date'][1], '%Y-%m-%d'))
 
-        vocab = []
-        for result in sqs:
-            if result.title:
-                vocab.extend(result.title.split(' '))
-            if result.subject:
-                vocab.extend(result.subject)
+                    # (datefilter.start < start_date < datefilter.end) or (start_date < datefilter.start)
+                    sqs = sqs.exclude(start_date__gt=datefilter.end).exclude(end_date__lt=datefilter.start)
+                except ValueError as e:
+                    pass  # ignore bad values and don't filter which is the correct action
```

Comment:
So you ignore a filter when the request parameters are bad?  I'm not sure I agree that is the correct action.  If the dates don't make sense (start_date is later than end_date) then a response with a bad request code makes sense to me.  If the parameters are valid but don't match any results, then a response with an empty result set makes sense.  Maybe I don't understand what is going on here?

Reply:
It was expecting two dates and dropping if it didn't have them. We discussed this on the frontend and are refactoring, anyway. The new behavior is to error with malformed requests, but the frontend is doing data validation, so if you ask for something unrealistic, you get no results. "There are no dates between 2010 start date and 2009 end date", so no resources are returned. Commit 964479979